### ISSUE TITLE
Early stopping on testing set

### ISSUE THREAD

**Issue by nguyenkhoa0209** [lululxvi/deepxde#166]
Dear Dr. Lu,

I read the docs about _EarlyStopping_ in the _callbacks_ and it is said that "Stop training when a monitored quantity (training loss) has stopped improving."

Is there any way to use early stopping in the testing data set? For example in the code below (from the Burger.py), I specify the testing set by adding `num_test=1000`. How could I use early stopping on this 1000 points testing data set? 

Furthermore, is there any way that I can choose these 1000 points randomly? As I understand in your code data/pde.py, the testing points are uniformly distributed.  

Thank you for your help.

```
data_xde = dde.data.TimePDE(
    geomtime, pde, [bc, ic], num_domain=2000, num_boundary=20, num_initial=80,  num_test=1000
)
net = dde.maps.FNN([2] + [20] * 3 + [1], "tanh", "Glorot normal")
model = dde.Model(data_xde, net)
early_stopping = dde.callbacks.EarlyStopping(min_delta=1e-6, patience=5000)
model.compile("adam", lr=1.0e-3)
model.train(epochs=15000, callbacks=[early_stopping])`

**Comment by lululxvi** [comment]
To use early stopping in the test dataset, you have to modify `loss_train` to `loss_test` in the code
https://github.com/lululxvi/deepxde/blob/a92ca8a3a7a8eaa9316e848f169d2bce4e7bdc7a/deepxde/callbacks.py#L211

To use random test data, you need to modify
https://github.com/lululxvi/deepxde/blob/a92ca8a3a7a8eaa9316e848f169d2bce4e7bdc7a/deepxde/data/pde.py#L192

**Comment by nguyenkhoa0209** [comment]
Thanks a lot for your answer. I'll close this issue.
